### ISSUE TITLE
not working for matplotlib, missing fix for libGL/libstdc

### ISSUE THREAD

**Issue by bileamScheuvens** [GuillaumeDesforges/fix-python#19]
running fix-python in an environment with matplotlib patches PyQt5/6 but attempting to plot shows:
```Python
import matplotlib.pyplot as plt
plt.figure()
plt.show()
>> "UserWarning: FigureCanvasAgg is non-interactive, and thus cannot be shown"

from PyQt5.QtWidgets import *
app = QApplication([])
win = QMainWindow()
win.show()
app.exec()
>> "ImportError: libGL.so.1: cannot open shared object file: No such file or directory"

from PyQt6.QtWidgets import *
app = QApplication([])
win = QMainWindow()
win.show()
app.exec()
>> "ImportError: libstdc++.so.6: cannot open shared object file: No such file or directory"
```

**Comment by GuillaumeDesforges** [comment]
I can reproduce

```console
$ cd $(mktemp -d)
$ nix run nixpkgs#python3 -- -m venv .venv --copies
$ source .venv/bin/activate
$ pip install matplotlib
$ pip install PyQt6
$ nix run github:GuillaumeDesforges/fix-python -- --venv .venv
$ python
>>> from PyQt6.QtWidgets import *
ImportError: libstdc++.so.6: cannot open shared object file: No such file or directory
```

**Comment by GuillaumeDesforges** [comment]
With `LD_DEBUG=libs` on my machine (not minimal)

```
>>> from PyQt6.QtWidgets import *
     15892:	find library=libQt6Widgets.so.6 [0]; searching
     15892:	search path=/etc/sane-libs:/nix/store/pbvn150v0w5v1jadv1b8wnvkp81vxz5k-pipewire-1.2.6-jack/lib		(LD_LIBRARY_PATH)
     15892:	 trying file=/etc/sane-libs/libQt6Widgets.so.6
     15892:	 trying file=/nix/store/pbvn150v0w5v1jadv1b8wnvkp81vxz5k-pipewire-1.2.6-jack/lib/libQt6Widgets.so.6
     15892:	search path=/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/lib/glibc-hwcaps/x86-64-v4:/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/lib/glibc-hwcaps/x86-64-v3:/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/lib/glibc-hwcaps/x86-64-v2:/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/lib		(RUNPATH from file python)
     15892:	 trying file=/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/lib/glibc-hwcaps/x86-64-v4/libQt6Widgets.so.6
     15892:	 trying file=/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/lib/glibc-hwcaps/x86-64-v3/libQt6Widgets.so.6
     15892:	 trying file=/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/lib/glibc-hwcaps/x86-64-v2/libQt6Widgets.so.6
     15892:	 trying file=/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/lib/libQt6Widgets.so.6
     15892:	
     15892:	find library=libQt6Gui.so.6 [0]; searching
     15892:	search path=/etc/sane-libs:/nix/store/pbvn150v0w5v1jadv1b8wnvkp81vxz5k-pipewire-1.2.6-jack/lib		(LD_LIBRARY_PATH)
     15892:	 trying file=/etc/sane-libs/libQt6Gui.so.6
     15892:	 trying file=/nix/store/pbvn150v0w5v1jadv1b8wnvkp81vxz5k-pipewire-1.2.6-jack/lib/libQt6Gui.so.6
     15892:	search path=/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/lib		(RUNPATH from file python)
     15892:	 trying file=/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/lib/libQt6Gui.so.6
     15892:	
     15892:	find library=libQt6Core.so.6 [0]; searching
     15892:	search path=/etc/sane-libs:/nix/store/pbvn150v0w5v1jadv1b8wnvkp81vxz5k-pipewire-1.2.6-jack/lib		(LD_LIBRARY_PATH)
     15892:	 trying file=/etc/sane-libs/libQt6Core.so.6
     15892:	 trying file=/nix/store/pbvn150v0w5v1jadv1b8wnvkp81vxz5k-pipewire-1.2.6-jack/lib/libQt6Core.so.6
     15892:	search path=/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/lib		(RUNPATH from file python)
     15892:	 trying file=/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/lib/libQt6Core.so.6
     15892:	
     15892:	find library=libstdc++.so.6 [0]; searching
     15892:	search path=/etc/sane-libs:/nix/store/pbvn150v0w5v1jadv1b8wnvkp81vxz5k-pipewire-1.2.6-jack/lib		(LD_LIBRARY_PATH)
     15892:	 trying file=/etc/sane-libs/libstdc++.so.6
     15892:	 trying file=/nix/store/pbvn150v0w5v1jadv1b8wnvkp81vxz5k-pipewire-1.2.6-jack/lib/libstdc++.so.6
     15892:	search path=/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/lib		(RUNPATH from file python)
     15892:	 trying file=/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/lib/libstdc++.so.6
     15892:	search cache=/nix/store/wn7v2vhyyyi6clcyn0s9ixvl7d4d87ic-glibc-2.40-36/etc/ld.so.cache
     15892:	search path=/nix/store/wn7v2vhyyyi6clcyn0s9ixvl7d4d87ic-glibc-2.40-36/lib:/nix/store/2d5spnl8j5r4n1s4bj1zmra7mwx0f1n8-xgcc-13.3.0-libgcc/lib/glibc-hwcaps/x86-64-v4:/nix/store/2d5spnl8j5r4n1s4bj1zmra7mwx0f1n8-xgcc-13.3.0-libgcc/lib/glibc-hwcaps/x86-64-v3:/nix/store/2d5spnl8j5r4n1s4bj1zmra7mwx0f1n8-xgcc-13.3.0-libgcc/lib/glibc-hwcaps/x86-64-v2:/nix/store/2d5spnl8j5r4n1s4bj1zmra7mwx0f1n8-xgcc-13.3.0-libgcc/lib		(system search path)
     15892:	 trying file=/nix/store/wn7v2vhyyyi6clcyn0s9ixvl7d4d87ic-glibc-2.40-36/lib/libstdc++.so.6
     15892:	 trying file=/nix/store/2d5spnl8j5r4n1s4bj1zmra7mwx0f1n8-xgcc-13.3.0-libgcc/lib/glibc-hwcaps/x86-64-v4/libstdc++.so.6
     15892:	 trying file=/nix/store/2d5spnl8j5r4n1s4bj1zmra7mwx0f1n8-xgcc-13.3.0-libgcc/lib/glibc-hwcaps/x86-64-v3/libstdc++.so.6
     15892:	 trying file=/nix/store/2d5spnl8j5r4n1s4bj1zmra7mwx0f1n8-xgcc-13.3.0-libgcc/lib/glibc-hwcaps/x86-64-v2/libstdc++.so.6
     15892:	 trying file=/nix/store/2d5spnl8j5r4n1s4bj1zmra7mwx0f1n8-xgcc-13.3.0-libgcc/lib/libstdc++.so.6
     15892:	
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: libstdc++.so.6: cannot open shared object file: No such file or directory
```

but `ldd` does show libstdc++ for all qt related binaries

```
$ find "$(realpath ".venv")" -type f -executable -exec sh -c "file -i '{}' | grep -qE 'x-(.*); charset=binary'" \; -print | grep qt | xargs -I {} sh -c 'echo {} && (ldd {} | grep libstd)'
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/qml/QtQuick3D/Effects/libqtquick3deffectplugin.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007ff410a00000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/qml/QtQuick3D/Particles3D/libqtquick3dparticles3dplugin.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007f9865000000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/qml/QtQuick3D/ParticleEffects/libqtquick3dparticleeffectsplugin.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007f5050400000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/qml/QtQuick3D/Helpers/impl/libqtquick3dhelpersimplplugin.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007f233f800000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/qml/QtQuick3D/Helpers/libqtquick3dhelpersplugin.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007f3905600000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/qml/QtQuick3D/AssetUtils/libqtquick3dassetutilsplugin.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007f5528200000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/qml/QtQuick3D/Physics/Helpers/libqtquick3dphysicshelpersplugin.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007fdfba200000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/qml/QtQuick/libqtquick2plugin.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007fb8c9a00000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/qml/QtQuick/Dialogs/libqtquickdialogsplugin.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007fc2bc800000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/qml/QtQuick/Dialogs/quickimpl/libqtquickdialogs2quickimplplugin.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007ffa73200000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/qml/QtQuick/Scene2D/libqtquickscene2dplugin.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007f5381c00000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/qml/QtQuick/Controls/Basic/libqtquickcontrols2basicstyleplugin.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007f19efa00000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/qml/QtQuick/Controls/Basic/impl/libqtquickcontrols2basicstyleimplplugin.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007f2900e00000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/qml/QtQuick/Controls/Imagine/libqtquickcontrols2imaginestyleplugin.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007fb69e400000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/qml/QtQuick/Controls/Imagine/impl/libqtquickcontrols2imaginestyleimplplugin.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007f5146200000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/qml/QtQuick/Controls/Fusion/libqtquickcontrols2fusionstyleplugin.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007fe696800000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/qml/QtQuick/Controls/Fusion/impl/libqtquickcontrols2fusionstyleimplplugin.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007f822fa00000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/qml/QtQuick/Controls/Universal/impl/libqtquickcontrols2universalstyleimplplugin.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007ff153200000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/qml/QtQuick/Controls/Universal/libqtquickcontrols2universalstyleplugin.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007f526e200000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/qml/QtQuick/Controls/impl/libqtquickcontrols2implplugin.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007f2c1aa00000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/qml/QtQuick/Controls/libqtquickcontrols2plugin.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007f769ec00000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/qml/QtQuick/Controls/Material/libqtquickcontrols2materialstyleplugin.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007f7f5b400000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/qml/QtQuick/Controls/Material/impl/libqtquickcontrols2materialstyleimplplugin.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007f091e200000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/qml/QtQuick/Timeline/BlendTrees/libqtquicktimelineblendtreesplugin.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007f6333600000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/qml/QtQuick/Timeline/libqtquicktimelineplugin.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007fea6ca00000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/qml/QtQuick/Templates/libqtquicktemplates2plugin.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007f1411800000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/qml/QtQuick/Scene3D/libqtquickscene3dplugin.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007f8b29600000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/qml/QtQuick/NativeStyle/libqtquickcontrols2nativestyleplugin.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007ff397e00000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/plugins/webview/libqtwebview_webengine.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007f08fc800000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/plugins/wayland-shell-integration/libqt-shell.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007f7f13000000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/plugins/imageformats/libqtiff.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007f8394800000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/plugins/imageformats/libqtga.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007f83b2200000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/plugins/position/libqtposition_positionpoll.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007efdc5e00000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/plugins/position/libqtposition_geoclue2.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007f10a4800000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/plugins/position/libqtposition_nmea.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007f806c800000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/plugins/generic/libqtuiotouchplugin.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007f2f20000000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/plugins/texttospeech/libqtexttospeech_speechd.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007f14f2600000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/plugins/texttospeech/libqtexttospeech_mock.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007f9959400000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/plugins/sensors/libqtsensors_iio-sensor-proxy.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007fb506400000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/plugins/sensors/libqtsensors_generic.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007ff627800000)
/tmp/tmp.Bcsi1DXuhO/.venv/lib/python3.12/site-packages/PyQt6/Qt6/plugins/wayland-graphics-integration-client/libqt-plugin-wayland-egl.so
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007f8111a00000)
```

**Comment by GuillaumeDesforges** [comment]
Found the culprit with `LD_DEBUG=all`

```
     11897:	file=libstdc++.so.6 [0];  needed by /home/gdforj/tmp/python/.venv/lib/python3.12/site-packages/PyQt6/QtWidgets.abi3.so [0]
     11897:	find library=libstdc++.so.6 [0]; searching
     11897:	 search path=/etc/sane-libs:/nix/store/pbvn150v0w5v1jadv1b8wnvkp81vxz5k-pipewire-1.2.6-jack/lib		(LD_LIBRARY_PATH)
     11897:	  trying file=/etc/sane-libs/libstdc++.so.6
     11897:	  trying file=/nix/store/pbvn150v0w5v1jadv1b8wnvkp81vxz5k-pipewire-1.2.6-jack/lib/libstdc++.so.6
     11897:	 search path=/home/gdforj/tmp/python/.venv/lib/python3.12/site-packages/PyQt6/Qt6/lib		(RUNPATH from file python)
     11897:	  trying file=/home/gdforj/tmp/python/.venv/lib/python3.12/site-packages/PyQt6/Qt6/lib/libstdc++.so.6
     11897:	 search cache=/nix/store/sl141d1g77wvhr050ah87lcyz2czdxa3-glibc-2.40-36/etc/ld.so.cache
     11897:	 search path=/nix/store/sl141d1g77wvhr050ah87lcyz2czdxa3-glibc-2.40-36/lib:/nix/store/4hcdxyjf9yiq7qf3i4548drb6sjmwa1v-xgcc-13.3.0-libgcc/lib/glibc-hwcaps/x86-64-v4:/nix/store/4hcdxyjf9yiq7qf3i4548drb6sjmwa1v-xgcc-13.3.0-libgcc/lib/glibc-hwcaps/x86-64-v3:/nix/store/4hcdxyjf9yiq7qf3i4548drb6sjmwa1v-xgcc-13.3.0-libgcc/lib/glibc-hwcaps/x86-64-v2:/nix/store/4hcdxyjf9yiq7qf3i4548drb6sjmwa1v-xgcc-13.3.0-libgcc/lib		(system search path)
     11897:	  trying file=/nix/store/sl141d1g77wvhr050ah87lcyz2czdxa3-glibc-2.40-36/lib/libstdc++.so.6
     11897:	  trying file=/nix/store/4hcdxyjf9yiq7qf3i4548drb6sjmwa1v-xgcc-13.3.0-libgcc/lib/glibc-hwcaps/x86-64-v4/libstdc++.so.6
     11897:	  trying file=/nix/store/4hcdxyjf9yiq7qf3i4548drb6sjmwa1v-xgcc-13.3.0-libgcc/lib/glibc-hwcaps/x86-64-v3/libstdc++.so.6
     11897:	  trying file=/nix/store/4hcdxyjf9yiq7qf3i4548drb6sjmwa1v-xgcc-13.3.0-libgcc/lib/glibc-hwcaps/x86-64-v2/libstdc++.so.6
     11897:	  trying file=/nix/store/4hcdxyjf9yiq7qf3i4548drb6sjmwa1v-xgcc-13.3.0-libgcc/lib/libstdc++.so.6
     11897:	
     11897:	
     11897:	file=/home/gdforj/tmp/python/.venv/lib/python3.12/site-packages/PyQt6/QtWidgets.abi3.so [0];  destroying link map
     11897:	
     11897:	file=/home/gdforj/tmp/python/.venv/lib/python3.12/site-packages/PyQt6/Qt6/lib/libQt6Widgets.so.6 [0];  destroying link map
     11897:	
     11897:	file=/home/gdforj/tmp/python/.venv/lib/python3.12/site-packages/PyQt6/Qt6/lib/libQt6Gui.so.6 [0];  destroying link map
     11897:	
     11897:	file=/home/gdforj/tmp/python/.venv/lib/python3.12/site-packages/PyQt6/Qt6/lib/libQt6Core.so.6 [0];  destroying link map
Traceback (most recent call last):
  File "/home/gdforj/tmp/python/test.py", line 1, in <module>
    from PyQt6.QtWidgets import *
ImportError: libstdc++.so.6: cannot open shared object file: No such file or directory
```

but

```
$ ldd .venv/lib/python3.12/site-packages/PyQt6/QtWidgets.abi3.so
ldd: warning: you do not have execution permission for `.venv/lib/python3.12/site-packages/PyQt6/QtWidgets.abi3.so'
	linux-vdso.so.1 (0x00007ffe203d3000)
	libQt6Widgets.so.6 => /home/gdforj/tmp/python/.venv/lib/python3.12/site-packages/PyQt6/Qt6/lib/libQt6Widgets.so.6 (0x00007fc15d000000)
	libQt6Gui.so.6 => /home/gdforj/tmp/python/.venv/lib/python3.12/site-packages/PyQt6/Qt6/lib/libQt6Gui.so.6 (0x00007fc15c400000)
	libQt6Core.so.6 => /home/gdforj/tmp/python/.venv/lib/python3.12/site-packages/PyQt6/Qt6/lib/libQt6Core.so.6 (0x00007fc15bc00000)
	libstdc++.so.6 => not found
	libc.so.6 => /nix/store/wn7v2vhyyyi6clcyn0s9ixvl7d4d87ic-glibc-2.40-36/lib/libc.so.6 (0x00007fc15ba07000)
	libGL.so.1 => not found
	libxkbcommon.so.0 => not found
	libpthread.so.0 => /nix/store/wn7v2vhyyyi6clcyn0s9ixvl7d4d87ic-glibc-2.40-36/lib/libpthread.so.0 (0x00007fc15de3e000)
	libstdc++.so.6 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libstdc++.so.6 (0x00007fc15b600000)
	libm.so.6 => /nix/store/wn7v2vhyyyi6clcyn0s9ixvl7d4d87ic-glibc-2.40-36/lib/libm.so.6 (0x00007fc15cf19000)
	libgcc_s.so.1 => /nix/store/ybjcla5bhj8g1y84998pn4a2drfxybkv-gcc-13.3.0-lib/lib/libgcc_s.so.1 (0x00007fc15de17000)
	libEGL.so.1 => not found
	libfontconfig.so.1 => not found
	libX11.so.6 => not found
	libdl.so.2 => /nix/store/wn7v2vhyyyi6clcyn0s9ixvl7d4d87ic-glibc-2.40-36/lib/libdl.so.2 (0x00007fc15de10000)
	libglib-2.0.so.0 => not found
	libQt6DBus.so.6 => /home/gdforj/tmp/python/.venv/lib/python3.12/site-packages/PyQt6/Qt6/lib/libQt6DBus.so.6 (0x00007fc15c327000)
	libxkbcommon.so.0 => not found
	libGL.so.1 => not found
	libz.so.1 => /nix/store/bzk3q2l71qwhycsip23y6rl5n881la4n-zlib-1.3.1/lib/libz.so.1 (0x00007fc15ddf0000)
	libfreetype.so.6 => not found
	libgthread-2.0.so.0 => not found
	libicui18n.so.73 => /home/gdforj/tmp/python/.venv/lib/python3.12/site-packages/PyQt6/Qt6/lib/libicui18n.so.73 (0x00007fc15b200000)
	libicuuc.so.73 => /home/gdforj/tmp/python/.venv/lib/python3.12/site-packages/PyQt6/Qt6/lib/libicuuc.so.73 (0x00007fc15ae00000)
	libicudata.so.73 => /home/gdforj/tmp/python/.venv/lib/python3.12/site-packages/PyQt6/Qt6/lib/libicudata.so.73 (0x00007fc158e00000)
	libzstd.so.1 => not found
	libglib-2.0.so.0 => not found
	libgthread-2.0.so.0 => not found
	librt.so.1 => /nix/store/wn7v2vhyyyi6clcyn0s9ixvl7d4d87ic-glibc-2.40-36/lib/librt.so.1 (0x00007fc15dde9000)
	/nix/store/wn7v2vhyyyi6clcyn0s9ixvl7d4d87ic-glibc-2.40-36/lib64/ld-linux-x86-64.so.2 (0x00007fc15de47000)
	libdbus-1.so.3 => not found
```

`libstdc++.so.6` is twice in there, which I have not seen before.

EDIT:
```
$ patchelf --print-rpath .venv/lib/python3.12/site-packages/PyQt6/QtWidgets.abi3.so
$ORIGIN/Qt6/lib
```
seems like `fix-python` does not patch RPATH for this file. Let me look into it.

**Comment by GuillaumeDesforges** [comment]
Fixed in e48a60c2d6a888ad2c5ddfcf36590c924f5df58a

Please follow instructions in the README to create a `libs.nix` files for the `--libs` option.

For instance for libGL:

`libs.nix`

```nix
let
  pkgs = import (builtins.getFlake "nixpkgs") { };
in
[
  pkgs.gcc.cc
  pkgs.glibc
  pkgs.zlib
  pkgs.libGL
]
```

Note that you may need to add other libraries (and figure out which ones need to be added).
If you do that, it would be helpful if you could share the `libs.nix` with the set of dependencies required for PyQt6.

Let me know how it goes :slightly_smiling_face:

**Comment by bileamScheuvens** [comment]
Thank you so much!
For PyQt6 and by extension matplotlib i needed:

```libs.nix```
```nix
let
  pkgs = import (builtins.getFlake "nixpkgs") {};
in
  [
    pkgs.gcc.cc
    pkgs.glibc
    pkgs.zlib
    pkgs.libGL

    # additional
    pkgs.libxkbcommon
    pkgs.fontconfig
    pkgs.xorg.libX11
    pkgs.glib
    pkgs.freetype
    pkgs.zstd
    pkgs.dbus
    pkgs.xcb-util-cursor
    pkgs.wayland
  ]
```
In no specific order, just working from crash to crash.
None of these need to be patched, they work fine installed in the global config and not passed via libs.
The displaymanager stuff is platform specific, hope it helps nonetheless.

**Comment by GuillaumeDesforges** [comment]
For the record: the way to fix this issue is now to use the `--deep` flag. See https://github.com/GuillaumeDesforges/fix-python/issues/20#issuecomment-2593927221
